misc/extension_api_validation/4.3-stable.expected

```diff
@@ -122,3 +122,19 @@ GH-98918
 Validate extension JSON: Error: Field 'classes/FileAccess/methods/open_encrypted/arguments': size changed value in new API, from 3 to 4.
 
 Optional argument added to allow setting initialization vector. Compatibility method registered.
+
+
+GH-99341
+--------
+Validate extension JSON: Error: Field 'classes/PhysicsServer3D/methods/generic_6dof_joint_set_param/arguments/1': type changed value in new API, from "enum::Vector3.Axis" to "int".
+Validate extension JSON: Error: Field 'classes/PhysicsServer3D/methods/generic_6dof_joint_get_param/arguments/1': type changed value in new API, from "enum::Vector3.Axis" to "int".
```

Comment:
I'm not sure this is the right approach. Even if binary compatibility is retained, this changes the public API exposed towards extensions. 

Bindings such as godot-rust have been built around a type [Vector3Axis](https://godot-rust.github.io/docs/gdext/master/godot/classes/trait.IPhysicsServer3DExtension.html?search=vector3axis#tymethod.generic_6dof_joint_get_param), you can't just remove it because of an issue in `set_axis()` while it works everywhere else.

Reply:
If I may (as someone who noticed the issue but isn't proficient in C++), the issue seems to be deeper than just set_axis(), since this PR finds and eliminates references to the same enum in multiple classes. I've just checked & found that the enum [Vector3.Axis does exist](https://github.com/godotengine/godot/blob/master/core/math/vector3.h) in the code after all, but trying to access it in GDScript results in errors. If eliminating it breaks compat, then perhaps the right approach would be finding a way to expose it?

Reply:
> perhaps the right approach would be finding a way to expose it?

I thought the same but no class has exposed enums, so I thought that wasn't the intended approach.

Reply:
> perhaps the right approach would be finding a way to expose it?

I thought the same but no class has exposed enums, so I thought that wasn't the intended approach.

Reply:
> perhaps the right approach would be finding a way to expose it?

I thought the same but no class has exposed enums, so I thought that wasn't the intended approach.

Reply:
Precisely my thoughts. These enums from Variants should be exposed appropriately and used across the codebase like `Vector3.Axis` is. However, doing so is not straightforward because, unlike enums from classes inheriting Object, there's currently no way to do it.

Reply:
> I thought the same but no class has exposed enums, so I thought that wasn't the intended approach.
Some low-level classes have them; off the top of my head - [HTTPClient](https://docs.godotengine.org/en/4.3/classes/class_httpclient.html#class-httpclient) has enums for Method, Status and ResponseCode.

Reply:
I looked at other implementations for other classes like `Vector4` and `Vector2`. None of them had that issue, instead they treated `Vector2.Axis` params and return types as an `int`. At least for binded methods. So I thought I'd change any binded method that uses `Vector3.Axis` as param/return type to `int`.

I basically matched `Vector3`'s implementation with `Vector2`, `Vector2I`, `Vector3I`, `Vector4`, `Vector4I`, and `Plane`

Reply:
> Precisely my thoughts. These enums from Variants should be exposed appropriately and used across the codebase like `Vector3.Axis` is. However, doing so is not straightforward because, unlike enums from classes inheriting Object, there's currently no way to do it.

There are [global enums](https://docs.godotengine.org/en/stable/classes/class_@globalscope.html#enumerations) though. They already contain enums such as `EulerOrder` specific to geometric builtin types, so `Vector3Axis` and future `Vector2Axis/Vector4Axis` wouldn't be _that_ out of place.

It would still technically be a compatibility breakage, but:
- There's clear continuity; the binding just needs to understand both `Vector3.Axis` and `Vector3Axis` in the JSON.
- There's no downgrade in type safety (int may work for GDScript but is a non-starter for other languages).

Reply:
> > Precisely my thoughts. These enums from Variants should be exposed appropriately and used across the codebase like `Vector3.Axis` is. However, doing so is not straightforward because, unlike enums from classes inheriting Object, there's currently no way to do it.
> 
> There are [global enums](https://docs.godotengine.org/en/stable/classes/class_@globalscope.html#enumerations) though. They already contain enums such as `EulerOrder` specific to geometric builtin types, so `Vector3Axis` and future `Vector2Axis/Vector4Axis` wouldn't be _that_ out of place.
> 
> It would still technically be a compatibility breakage, but we don't just plain _remove_ a valid feature.
> 
> * There's clear continuity; the binding just needs to understand both `Vector3.Axis` and `Vector3Axis` in the JSON.
> * There's no downgrade in type safety (int may work for GDScript but is a non-starter for other languages).

What if we just deprecated VectorX.Axis enums and replaced them with `@GlobalScope.VectorAxis` that would looks something like this

```gdscript
VectorAxis {
    VECTOR_AXIS_X,
    VECTOR_AXIS_Y,
    VECTOR_AXIS_Z,
    VECTOR_AXIS_W
}
```

that way its only one enum for all vector types

Reply:
Not sure about a single enum type; you could then pass `VECTOR_AXIS_W` to an API that expects `Vector2.Axis`. If all APIs refer to `VectorAxis`, the documentation aspect (which dimension) also gets lost.